File: 4355-rearrange-string-to-avoid-character-pair/rearrange-string-to-avoid-character-pair.go
func rearrangeString(s string, x byte, y byte) string {
	n := len(s)
	var left, mid, right []byte

	for i := 0; i < n; i++ {
		if s[i] == y {
			left = append(left, s[i])
		} else if s[i] == x {
			right = append(right, s[i])
		} else {
			mid = append(mid, s[i])
		}
	}

	return string(append(append(left, mid...), right...))
}

File: 4355-rearrange-string-to-avoid-character-pair/rearrange-string-to-avoid-character-pair.py
class Solution:
    def rearrangeString(self, s: str, x: str, y: str) -> str:
        ans = ""

        for c in s:
            if c == y:
                ans += c
        for c in s:
            if c != x and c != y:
                ans += c
        for c in s:
            if c == x:
                ans += c

        return ans
